darts/models/forecasting/forecasting_model.py

```diff
@@ -1508,9 +1544,9 @@ def predict(
                 if isinstance(future_covariates.time_index, pd.DatetimeIndex)
                 else n
             )
-            future_covariates = future_covariates[
-                start : start + offset * self.training_series.freq
-            ]
+            future_covariates = future_covariates.slice(
```

Comment:
Don't forget that `slice()` is inclusive on the right for DatetimeIndex

Reply:
I think it is correct, but let me know if I'm missing something.
`start` is one step after end of target series, and we end at:

- n - 1 steps ahead of start for DatetimeIndex (as inclusive with slice)
- n steps ahead of start for RangeIndex (as non-inclusive with slice)